### PR TITLE
cherry picked PR of #10233

### PR DESCRIPTION
**Pull request by jmozah**
This PR adds a new validation where in the peer is not penalised (p2p peer scoring) if the blocks parent is optimistic.

check PR #10233 for the past review comments

### CODE REVIEW

**Review by terencechain — APPROVED** [review]
Lgtm. @nisdas please take a look as well

**Review by nisdas — APPROVED** [review]
This looks good to me, but reading the spec I realised that if bellatrix validation fails, we do not put the block in the bad block cache.

**Comment by jmozah** [comment]
> This looks good to me, but reading the spec I realised that if bellatrix validation fails, we do not put the block in the bad block cache.

@nisdas Is BadBlock concept not specific to impl.? But i agree that we should put it there to avoid DOS vectors. What is your opinion?

File: beacon-chain/sync/validate_beacon_blocks.go [hunk 1]
```diff
@@ -221,15 +230,26 @@ func (s *Service) validateBeaconBlock(ctx context.Context, blk block.SignedBeaco
 		return errors.New("incorrect proposer index")
 	}
 
-	return validateBellatrixBeaconBlock(parentState, blk.Block())
+	if err = s.validateBellatrixBeaconBlock(ctx, parentState, blk.Block()); err != nil {
+		s.setBadBlock(ctx, blockRoot)
```
Comment (nisdas) [inline beacon-chain/sync/validate_beacon_blocks.go:234]: we would set bad blocks for blocks with optimistic parents, so this edge case needs to be handled

**Review by nisdas — APPROVED** [review]

**Review by terencechain — APPROVED** [review]

**Pull request merged**